elasticgraph-warehouse/lib/elastic_graph/warehouse/schema_definition/index_extension.rb

```diff
@@ -19,25 +19,57 @@ module IndexExtension
         # @dynamic warehouse_table_def
         attr_reader :warehouse_table_def
 
-        # Defines a warehouse table for this index.
+        # Returns whether this index has been explicitly excluded from the data warehouse.
+        #
+        # @return [Boolean] `true` if {#exclude_from_warehouse} was called, `false` otherwise
+        def excluded_from_warehouse?
+          @excluded_from_warehouse == true
+        end
+
+        # Defines a warehouse table for this index with a custom name.
+        #
+        # By default, a warehouse table is automatically created with the same name as the index.
+        # Use this method only when you need a different table name than the index name.
+        # To exclude an index from the warehouse entirely, use {#exclude_from_warehouse} instead.
         #
         # @param name [String] name of the warehouse table
         # @return [void]
         #
-        # @example Define a warehouse table for the products index
+        # @example Override the default warehouse table name
         #   ElasticGraph.define_schema do |schema|
         #     schema.object_type "Product" do |t|
         #       t.field "id", "ID"
         #       t.field "name", "String"
         #
         #       t.index "products" do |i|
+        #         # Override to use a different table name than "products"
         #         i.warehouse_table "store_products"
         #       end
         #     end
         #   end
         def warehouse_table(name)
           @warehouse_table_def = WarehouseTable.new(name: name, index: self)
         end
+
+        # Excludes this index from the data warehouse configuration.
+        # This is useful when you have an index but don't want it to be included
+        # in the data warehouse.
+        #
+        # @return [void]
+        #
+        # @example Exclude an internal/test index from the warehouse
+        #   ElasticGraph.define_schema do |schema|
+        #     schema.object_type "InternalMetrics" do |t|
+        #       t.field "id", "ID"
+        #
+        #       t.index "internal_metrics" do |i|
+        #         i.exclude_from_warehouse  # This index won't be in the data warehouse
```

Comment:
```suggestion
        #         i.exclude_from_warehouse # This index won't be in the data warehouse
```

/nit (since we use one space before `#` for trailing comments everywhere else).